spec/events/badge.spec.js

```diff
@@ -66,17 +80,83 @@ describe('lib/events/badge', () => {
       }).toThrowError(/Trkref/);
     });
 
-    it('includes a JSON-encoded string of the options given', () => {
-      const expectedJsonString = JSON.stringify(badgeRenderedParams);
+    it('uses TRKREF from the root scope when not provided as argument', () => {
+      badgeRenderedParams.trkref = undefined;
+      badge = new Badge('MY_TRKREF');
+
+      badge.rendered(badgeRenderedParams);
+      expect(snowplow).toHaveBeenCalledWith(
+        jasmine.anything(),
+        jasmine.objectContaining({
+          data: jasmine.objectContaining({
+            trkref: 'MY_TRKREF',
+          }),
+        })
+      );
+    });
+
+    it('raises an error if ctaPageUse is not valid', () => {
+      badgeRenderedParams.ctaPageUse = 'NOT_A_REAL_PAGE_USE';
+
+      expect(() => {
+        badge.rendered(badgeRenderedParams);
+      }).toThrowError(/not included in the list/);
+    });
+
+    it('sorts and stringifies reviewable context', () => {
+      badge.rendered({ ...badgeRenderedParams, reviewableContext: { model: 'Focus', manufacturer: 'Ford' }});
+
+      expect(snowplow).toHaveBeenCalledWith(
+        jasmine.anything(),
+        jasmine.objectContaining({
+          data: jasmine.objectContaining({
+            reviewableContext: '{"manufacturer":"Ford","model":"Focus"}',
+          }),
+        })
+      );
+    });
+
+    it('determines CTA page use from badgeVariant', () => {
+      badgeRenderedParams.ctaPageUse = undefined;
+      badgeRenderedParams.badgeVariant = 'my_listing123';
```

Comment:
A variation of this same test with a `badgeVariant` name including `category` instead of `listing` improves the coverage.